tests/test-helpers/src/counter_contract.rs

```diff
@@ -0,0 +1,162 @@
+//! A simple contract that holds a counter in the storage slot zero.
+//! Every time the contract is called, it increments the counter by one and returns the new value.
+
+use fuel_core_client::client::{
+    types::TransactionStatus,
+    FuelClient,
+};
+use fuel_core_types::{
+    fuel_asm::{
+        op,
+        GTFArgs,
+        RegId,
+    },
+    fuel_tx::{
+        Bytes32,
+        ContractId,
+        CreateMetadata,
+        Finalizable,
+        Input,
+        Output,
+        Receipt,
+        StorageSlot,
+        Transaction,
+        TransactionBuilder,
+    },
+    fuel_types::BlockHeight,
+    fuel_vm::{
+        Salt,
+        SecretKey,
+    },
+};
+use futures::StreamExt;
+use rand::Rng;
+
+/// Deploy the contract
+pub async fn deploy(
+    client: &FuelClient,
+    rng: &mut rand::rngs::StdRng,
+) -> (BlockHeight, ContractId) {
+    let maturity = Default::default();
+
+    let chain_info = client.chain_info().await.expect("failed to get chain info");
+    let base_asset_id = chain_info.consensus_parameters.base_asset_id();
+
+    let code: Vec<_> = [
+        // Make zero key
+        op::movi(0x12, 32),
+        op::aloc(0x12),
+        // Read value
+        op::srw(0x10, 0x11, 0x12),
+        // Increment value
+        op::addi(0x10, 0x10, 1),
+        // Write value
+        op::sww(0x12, 0x11, 0x10),
+        // Return new counter value
+        op::ret(0x10),
+    ]
```

Comment:
```suggestion
    let code: Vec<_> = [
        // Make zero key
        op::movi(0x12, 32),
        op::aloc(0x12),
        // Read value
        op::srw(0x10, 0x11, RegId::HP),
        // Increment value
        op::addi(0x10, 0x10, 1),
        // Write value
        op::sww(RegId::HP, 0x11, 0x10),
        // Return new counter value
        op::ret(0x10),
    ]
```

Reply:
This is a bug as well, although it doesn't affect the test. Would be caught when https://github.com/FuelLabs/fuel-core/pull/2682 is implemented, but I'll include the fix in https://github.com/FuelLabs/fuel-core/pull/2491 that also uses the counter contract.

Reply:
Fixed in 294ea09015